nflow-engine/src/main/java/io/nflow/engine/workflow/curated/MaintenanceWorkflowStarter.java

```diff
@@ -0,0 +1,46 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.service.MaintenanceConfiguration;
+import io.nflow.engine.service.WorkflowInstanceService;
+import io.nflow.engine.workflow.instance.WorkflowInstance;
+import org.springframework.context.event.ContextStartedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import static org.joda.time.Period.days;
+import static org.joda.time.Period.years;
+
+@Component
+public class MaintenanceWorkflowStarter {
+  protected final WorkflowInstanceService instanceService;
+  protected final boolean insertOnStartup;
+
+  public MaintenanceWorkflowStarter(Environment env, WorkflowInstanceService instanceService) {
+    this.instanceService = instanceService;
+    this.insertOnStartup = env.getProperty("nflow.maintenance.start", Boolean.class, false);
```

Comment:
I think the autoStart or auto_start was better. How about enabled? It might be nice to able to disable the workflow via config also after it has been inserted once. This would of course require that the workflow itself supports this property.

Reply:
With the normal scenario of multiple executors running the disable would need to be marked into the database (for example by clearing nextActivation). But it would be confusing that setting enable=false on one node would effect the others.
I now renamed this to _insertWorkflowIfMissing_.

Reply:
True. Maybe the maintenance workflow could support a signal to skip the cleanup. If enabled, insert if not exists and clear the signal if it does. If not enabled, set the signal. Setting different values on different nodes would be strange anyway?

Reply:
Also: use getRequiredProperty and move the default to engine.properties